src/graphics/gfx_background.cpp

```diff
@@ -46,7 +46,7 @@ void DrawBackground(int S, int Z)
     A = 1; // Blocks
     if(Background2[S] == 1)
     {
-        int tempVar = static_cast<int>(floor(static_cast<double>((level[S].Width - level[S].X) / GFXBackground2Width[A] * 0.5))) + 1;
+        int tempVar = static_cast<int>(floor(static_cast<double>((level[S].Width - level[S].X) / GFXBackground2Width[A] * 0.5 + (double)ScreenW / GFXBackground2Width[A]))) + 1;
```

Comment:
btw, this monstrous `static_cast<int>(floor(static_cast<double>(....))` is the result of the VB to C++ mess, you may replace them with just an int cast.